Xamarin.Essentials/Geolocation/Geolocation.ios.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using CoreLocation;
+using Foundation;
+
+namespace Xamarin.Essentials
+{
+    public static partial class Geolocation
+    {
+        internal static bool IsSupported
+            => CLLocationManager.LocationServicesEnabled;
+
+        static async Task<Location> PlatformLastKnownLocationAsync()
+        {
+            await Permissions.RequireAsync(PermissionType.LocationWhenInUse);
+
+            var manager = new CLLocationManager();
+            var location = manager.Location;
+
+            return location.ToLocation();
+        }
+
+        static async Task<Location> PlatformLocationAsync(GeolocationRequest request, CancellationToken cancellationToken)
+        {
+            await Permissions.RequireAsync(PermissionType.LocationWhenInUse);
+
+            var tcs = new TaskCompletionSource<CLLocation>();
+
+            // the location manager requires an active run loop
+            // so just use the main loop
+            Platform.BeginInvokeOnMainThread(() =>
+            {
+                var manager = new CLLocationManager();
```

Comment:
We create and start the location manger on the main thread:
 - because we need an active run loop
 - it is non-blocking
 - it does not affect the actual call threading system